pages/insight_menu.php

```diff
@@ -35,27 +36,36 @@ function echoCategory($categoryId, $size = 1) {
             ?><span id='categoryPrice'><?= $category['Price'] ?></span><?php
         } ?>
     </h<?= $size ?>> 
+
 <!-- Echo category description -->
      <i><p>
         <?= $category['Description'] ?>
     </p></i> 
 
 <?php
+
 // If there are still subcategories the function will keep being called upon
-    if (!empty($subcategories)) {
+$dishes = base_query("SELECT * FROM Dish WHERE Category = :categoryId ORDER BY Position", [':categoryId' => $categoryId])->fetchAll();
+    if ($subcategories == true) {?>
```

Comment:
$subcategories is een array en zal daarom nooit true zijn.

Je kunt de if weg laten en gewoon eerst de gerechten echo-en en daarna kijken of er nog subcategories zijn.

Reply:
Dan krijg je dus zoiets:

```php
$dishes = // Haal op uit database
foreach ($dishes as $dish) {
   // Echo dish zoals normaal;
}

foreach ($subcategories as $category) {
   echoCategory($categoryId)
}
```